File: backend/src/integrations/unbrowse.py
import httpx
from src.core.config import settings

async def unbrowse_resolve(intent: str, url: str | None = None) -> dict:
    """Resolve an intent via Unbrowse. Falls back to raising an exception if unavailable."""
    async with httpx.AsyncClient(timeout=30) as client:
        payload = {
            "intent": intent,
            "params": {},
        }
        if url:
            payload["params"]["url"] = url
            payload["context"] = {"url": url}

        response = await client.post(
            f"{settings.unbrowse_url}/v1/intent/resolve",
            json=payload,
        )
        response.raise_for_status()
        return response.json()

File: frontend/src/lib/integrations/unbrowse.ts
import { config } from "../config";

export async function unbrowseResolve(
  intent: string,
  url?: string
): Promise<Record<string, unknown>> {
  const payload: Record<string, unknown> = {
    intent,
    params: {},
  };
  if (url) {
    (payload.params as Record<string, unknown>).url = url;
    payload.context = { url };
  }

  const response = await fetch(`${config.unbrowseUrl}/v1/intent/resolve`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(payload),
  });
  if (!response.ok) throw new Error(`Unbrowse error: ${response.status}`);
  return response.json();
}
